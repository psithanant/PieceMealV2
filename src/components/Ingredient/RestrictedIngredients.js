import React, { Component } from 'react';
import { ListView, TouchableWithoutFeedback, View } from 'react-native';
import { Button, ListItem, Text, Icon } from 'native-base';

export default class CheckBoxExample extends Component {

  loadDataSource() {
      const ds = new ListView.DataSource({
          rowHasChanged: (r1, r2) => r1 !== r2
      });

      this.dataSource = ds.cloneWithRows(this.props.list);
  }

  render() {
    this.loadDataSource();
    return (
      <TouchableWithoutFeedback
        onPress={() => this.onPress()}
      >
        <View>
          <ListView
            dataSource={this.dataSource}
            enableEmptySections
            renderRow={
              (rowData) =>
                <ListItem style={styles.containerStyle}>
                  <Button primary small>
                    <Icon name='beer' />
                  </Button>
                  <Text>{rowData.name}</Text>
                </ListItem>}
          />
        </View>
      </TouchableWithoutFeedback>
    );
  }
}

const styles = {
  containerStyle: {
    borderBottomWidth: 1,
    borderColor: '#ddd'
  },
  expandedContainerStyle: {
    backgroundColor: '#e5e5e5',
    borderBottomWidth: 1,
    borderColor: '#ccc'
  },
  nameStyle: {
    fontSize: 18,
    paddingLeft: 15
  }
};
