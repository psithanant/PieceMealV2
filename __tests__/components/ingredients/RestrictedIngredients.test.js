import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RestrictedIngredients from '../../../src/components/Ingredient/RestrictedIngredients';

describe('Resricted Ingredients', () => {
  it('renders RestrictedIngredients component', () => {
  const component = shallow(
    <RestrictedIngredients
      list={[]}
    />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
