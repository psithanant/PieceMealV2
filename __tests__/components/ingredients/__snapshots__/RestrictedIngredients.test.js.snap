// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resricted Ingredients renders RestrictedIngredients component 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <View>
    <ListViewMock
      dataSource={
        ListViewDataSource {
          "items": 0,
        }
      }
      enableEmptySections={true}
      renderRow={[Function]}
      renderScrollComponent={[Function]}
    />
  </View>
</TouchableWithoutFeedback>
`;
